pulpcore/tests/serializers/test_repository.py

```diff
@@ -0,0 +1,31 @@
+from unittest import TestCase
+
+from pulpcore.app.models import Distribution
+from pulpcore.app.serializers import DistributionSerializer
+
+
+class TestDistributinPath(TestCase):
+    def test_overlap(self):
+        Distribution.objects.create(base_path="foo/bar", name="foobar")
+        overlap_errors = {'base_path': ["Overlaps with existing distribution 'foobar'"]}
+
+        # test that the new distribution is nested in an existing
+        data = {"name": "foobarbaz", "base_path": "foo/bar/baz"}
+        serializer = DistributionSerializer(data=data)
+        self.assertFalse(serializer.is_valid())
+        self.assertDictEqual(overlap_errors, serializer.errors)
+
+        # test that the new distribution nests an existing
+        data = {"name": "foo", "base_path": "foo"}
+        serializer = DistributionSerializer(data=data)
+        self.assertFalse(serializer.is_valid())
+        self.assertDictEqual(overlap_errors, serializer.errors)
+
+    def test_no_overlap(self):
+        Distribution.objects.create(base_path="fu/bar", name="fubar")
+        Distribution.objects.create(base_path="fu", name="fu")
+
+        data = {"name": "fufu", "base_path": "fubar"}
+        serializer = DistributionSerializer(data=data)
+        self.assertTrue(serializer.is_valid())
+        self.assertDictEqual({}, serializer.errors)
```

Comment:
There are a few other cases I'd like to see here

1) Identical base path "foo/bar"
2) Trailing slash difference "foo/bar/"
3) Common base, different path "foo/legal"

Reply:
I think this is outside the scope of this PR. This PR only adds validation for overlapping paths.

Reply:
Feel free to merge without, bug IMO those would count as "overlapping"

Reply:
So 1 would be a violation of our uniq constraint and I'm not sure it would hit this overlapping validation. The same thing goes for two only that we validate slashes. 

Test 3 is actually an overlapping case. Will add a unit test for it.